docs/source-2.0/guides/client-guidance/identity-auth/index.md

```diff
@@ -0,0 +1,416 @@
+authenticating the client's identity. This section describes how to model the
+workflow of identity retrieval and request authentication in Smithy clients.
+
+## Identity
+
+An Identity is an entity representing **who the caller is**. Abstractly, the
+caller's identity could be anonymous, a token, a public/private key, etc.
+
+Identity types in the client runtime should typically extend from a base
+interface that exposes a single value: the expiration time of the identity. The
+API should then branch per auth scheme. Sometimes an Identity type may be
+re-used across multiple schemes.
+
+```java
+public interface Identity {
+	Instant expiration();
+}
+
+// for smithy.api#bearerTokenAuth
+public interface BearerTokenIdentity extends Identity {
+	String token();
+}
+
+// for aws.auth#sigv4, aws.auth#sigv4a
+public interface SigV4CredentialIdentity extends Identity {
+	String accessKeyId();
+	String secretAccessKey();
+	String sessionToken();
+}
+```
+
+## IdentityResolver
+
+An IdentityResolver implements the retrieval of a single type of identity.
+
+The IdentityResolver interface should be modeled as follows:
+
+```java
+public interface IdentityResolver<TIdentity extends Identity> {
+    TIdentity resolve(Object properties);
+}
+```
+
+There may be multiple valid implementations of an identity resolver for a given
+type. For example, one implementation of an IdentityResolver<BearerTokenIdentity>
+may source the bearer token from the system environment, while another may
+source the token from in-code configuration.
+
+The client configuration for a service should typically include a field that
+allows configuration of an IdentityResolver for every auth scheme supported by
+the service.
+
+```java
+public class MyServiceClientConfig {
+    private final IdentityResolver<SigV4CredentialIdentity> sigV4CredentialResolver;
+    private final IdentityResolver<BearerTokenIdentity> bearerTokenResolver;
+
+    // other configuration...
+}
+```
+
+## Signer
+
+A Signer is an entity representing **a way to generate a signature for a
+request**. A signature is metadata attached to a request that will be sent to
+the service in order to allow the service to authenticate the Smithy client
+caller's identity.
+
+For example, an operation that uses a bearer token for
+identification would "sign" an outgoing HTTP request by presenting the value of
+the token in the Authorization header.
+
+All signers require, at a minimum, a request to sign and an identity with which
+to sign. A Signer should be modeled like so:
+
+```java
+// This example interface modifies the transport message in-place with the
+// signature. Depending on the surrounding runtime, the caller may instead wish to
+// model the Signer as an interface that returns a copied, modified transport
+// message instead.
+public interface Signer<TIdentity extends Identity, TMessage extends Message> {
+    void sign(TIdentity identity, TMessage message, Object properties);
```

Comment:
The original doc assumed HTTP, I know formally we want to speak in terms of transport agnosticism so I've parameterized the transport type.